clippy_lints/src/default.rs

```diff
@@ -81,6 +81,7 @@ impl<'tcx> LateLintPass<'tcx> for Default {
     fn check_expr(&mut self, cx: &LateContext<'tcx>, expr: &'tcx Expr<'_>) {
         if_chain! {
             if !expr.span.from_expansion();
+            if !is_update_syntax_base(cx, expr);
```

Comment:
We can discuss about the position of the check inside of the `if_chain`

Reply:
I would move it down to the point where we at least know that we're dealing with a `Default::default()` call. I don't think performance matters here, though I think the code is easier to read (because you know what `expr` is) if it is placed later in the `if_chain`